src/eddymotion/estimator.py

```diff
@@ -82,32 +82,47 @@ def fit(
         for i_iter in range(1, n_iter + 1):
             index_order = np.arange(len(dwdata))
             np.random.shuffle(index_order)
-            with tqdm(total=len(index_order), unit="dwi") as pbar:
-                for i in index_order:
+
+            dwmodel = None
+            single_model = model.lower() in ("b0", "s0", "avg", "average", "mean")
+            if single_model:
```

Comment:
In this mode, maybe we can predict in one fell swoop and avoid the loop below?

Reply:
We still need to run one registration per orientation, so I'm not sure predicting all orientations at once will give us any advantage. The cool thing is that we can now parallelize the loop in the `single_model` case.

This is why I was thinking of thread parallelization of the model fit/predict, and then use multiprocessing parallelization (e.g., joblib, or, better in this case, asyncio) for the loop.

Reply:
BTW, regarding the registration - @tmspvn had a pretty cool idea that would definitely move me away from ANTs (in favor of DIPY where this could be -I believe- implemented):

Very rarely, image registration iterations will be run on the full dataset. We could integrate fit/predict (for costly methods) within the cost function so that we don't have to model all voxels. To have an advantage, we will have the complication of caching modeled voxels as they will likely be picked at some later registration iteration.

WDTY?